core/blockchain/dbhelper.go

```diff
@@ -463,14 +463,14 @@ func dbRemoveBlockIndex(dbTx database.Tx, hash *hash.Hash, order int64) error {
 
 // dbPutBestState uses an existing database transaction to update the best chain
 // state with the given parameters.
-func dbPutBestState(dbTx database.Tx, snapshot *BestState, workSum *big.Int) error {
+func dbPutBestState(dbTx database.Tx, snapshot *BestState, workSum *big.Int, tokenTipID uint32) error {
 	// Serialize the current best chain state.
 	serializedData := serializeBestChainState(bestChainState{
 		hash:       snapshot.Hash,
 		total:      uint64(snapshot.GraphState.GetTotal()),
 		totalTxns:  snapshot.TotalTxns,
 		workSum:    workSum,
-		tokenTipID: snapshot.TokenTipID,
+		tokenTipID: tokenTipID,
```

Comment:
why it necessary to save `tokenTipID` into serialized data, instead of using the `tokenTipHash` ?
```go
func dbPutBestState(dbTx database.Tx, snapshot *BestState, workSum *big.Int) error {
   ... 
   tokenTipHash : snapshot.TokenTipHash
   ...
}
```

Reply:
Either way is OK